File: app/src/main/java/com/tendebit/dungeonmaster/charactercreation/classselection/model/CharacterClassInfo.java
package com.tendebit.dungeonmaster.charactercreation.classselection.model;

import com.google.gson.annotations.SerializedName;
import com.tendebit.dungeonmaster.charactercreation.proficiencyselection.model.CharacterProficiencyDirectory;
import com.tendebit.dungeonmaster.charactercreation.proficiencyselection.model.CharacterProficiencyGroup;
import com.tendebit.dungeonmaster.core.model.SelectionElement;

import org.jetbrains.annotations.NotNull;

import java.util.List;

@SuppressWarnings({"unused", "MismatchedQueryAndUpdateOfCollection"})
public class CharacterClassInfo implements SelectionElement {
    public String name;
    public String url;
    @SerializedName("hit_die")
    private int hitDie;
    @SerializedName("proficiency_choices")
    public List<CharacterProficiencyGroup> proficiencyChoices;
    private List<CharacterProficiencyDirectory> proficiencies;

    @Override
    public String toString() {
        return "\nInfo for class " + name + ": \n"
                + " Hit Die: " + hitDie + "\n"
                + " Natural Proficiencies: \n" + proficiencies + "\n"
                + " Proficiency Choices: \n" + proficiencyChoices;
    }

    @Override
    public boolean equals(Object obj) {
        return obj instanceof CharacterClassInfo && ((CharacterClassInfo) obj).name.equals(name);
    }

    @NotNull
    @Override
    public String primaryText() {
        return name;
    }

    @NotNull
    @Override
    public String primaryId() {
        return url;
    }
}

File: app/src/main/java/com/tendebit/dungeonmaster/charactercreation/pages/classselection/model/CharacterClassInfo.kt
package com.tendebit.dungeonmaster.charactercreation.pages.classselection.model

import com.google.gson.annotations.SerializedName
import com.tendebit.dungeonmaster.charactercreation.pages.proficiencyselection.model.CharacterProficiencyDirectory
import com.tendebit.dungeonmaster.charactercreation.pages.proficiencyselection.model.CharacterProficiencyGroup
import com.tendebit.dungeonmaster.core.model.SelectableElement

class CharacterClassInfo : SelectableElement {
    lateinit var name: String
    lateinit var url: String
    @SerializedName("hit_die")
    private val hitDie: Int = 0
    @SerializedName("proficiency_choices")
    lateinit var proficiencyChoices: List<CharacterProficiencyGroup>
    private val proficiencies: List<CharacterProficiencyDirectory>? = null

    override fun toString(): String {
        return ("\nInfo for class " + name + ": \n"
                + " Hit Die: " + hitDie + "\n"
                + " Natural Proficiencies: \n" + proficiencies + "\n"
                + " Proficiency Choices: \n" + proficiencyChoices)
    }

    override fun equals(other: Any?): Boolean {
        return other is CharacterClassInfo && other.name == name
    }

    override fun primaryText(): String {
        return name
    }

    override fun primaryId(): String {
        return url
    }

    override fun hashCode(): Int {
        return primaryId().hashCode()
    }
}